packages/astro/src/content/utils.ts

```diff
@@ -358,3 +387,90 @@ function search(fs: typeof fsMod, srcDir: URL) {
 	}
 	return { exists: false, url: paths[0] };
 }
+
+export async function getStringifiedLookupMap({
+	contentPaths,
+	contentEntryConfigByExt,
+	root,
+	fs,
+}: {
+	contentEntryConfigByExt: ReturnType<typeof getContentEntryConfigByExtMap>;
+	contentPaths: Pick<ContentPaths, 'contentDir' | 'cacheDir'>;
+	root: URL;
+	fs: typeof fsMod;
+}) {
+	const { contentDir } = contentPaths;
+	const globOpts: FastGlobOptions = {
+		absolute: true,
+		cwd: fileURLToPath(root),
+		fs: {
+			readdir: fs.readdir.bind(fs),
+			readdirSync: fs.readdirSync.bind(fs),
+		},
+	};
+
+	const relContentDir = rootRelativePath(root, contentDir, false);
+	const contentGlob = await glob(
+		`${relContentDir}**/*${getExtGlob([...contentEntryConfigByExt.keys()])}`,
+		globOpts
+	);
+	let filePathByLookupId: {
+		[collection: string]: Record<string, string>;
+	} = {};
+
+	await Promise.all(
+		contentGlob.map(async (filePath) => {
+			const info = getEntryInfo({ contentDir, entry: filePath });
+			if (info instanceof NoCollectionError) return;
+			const contentEntryType = contentEntryConfigByExt.get(extname(filePath));
+			if (!contentEntryType) return;
+
+			const { id, collection, slug: generatedSlug } = info;
+			filePathByLookupId[collection] ??= {};
+			const slug = await getEntrySlug({
+				id,
+				collection,
+				generatedSlug,
+				fs,
+				fileUrl: pathToFileURL(filePath),
+				contentEntryType,
+			});
+			filePathByLookupId[collection][slug] = rootRelativePath(root, filePath);
+		})
+	);
+
+	return JSON.stringify(filePathByLookupId);
+}
+
+/**
+ * Check for slug in content entry frontmatter and validate the type,
+ * falling back to the `generatedSlug` if none is found.
+ */
+export async function getEntrySlug({
+	id,
+	collection,
+	generatedSlug,
+	contentEntryType,
+	fileUrl,
+	fs,
+}: {
+	fs: typeof fsMod;
+	id: string;
+	collection: string;
+	generatedSlug: string;
+	fileUrl: URL;
+	contentEntryType: Pick<ContentEntryType, 'getEntryInfo'>;
+}) {
+	const { slug: frontmatterSlug } = await contentEntryType.getEntryInfo({
+		fileUrl,
+		contents: await fs.promises.readFile(fileUrl, 'utf-8'),
```

Comment:
If we can't read the file, an error will be fired. Do you know if we handle the error case already?

Reply:
Ah, this definitely assumed errors were an impossible case. Could throw an unknown content collection error maybe?

Reply:
Your assumptions are entirely valid! Although, you never know what the users/rights of the scripts that install/create the files/directory. 

My golden rule is to assume that when working with files, the code might need to have the rights proper rights to run the operation.